mii/deployment.py

```diff
@@ -160,6 +175,29 @@ def _deploy_aml(deployment_name, model_name, version):
     print("Please run 'deploy.sh' to bring your deployment online")
 
 
+def deploy_non_persistent(task,
+                          model,
+                          deployment_name,
+                          deployment_type=DeploymentType.LOCAL,
+                          model_path=None,
+                          enable_deepspeed=True,
+                          enable_zero=False,
+                          ds_config=None,
+                          mii_config={},
+                          version=1):
```

Comment:
If we introduce "deploy_non_persistent" which returns a client object, can we have an inference pipeline in the object instead of having it globally? We could tie the lifetimes of the client, an inference pipeline, and the deployment name.